### DISCUSSION TITLE
How to get the document title using id?

### DISCUSSION THREAD

**Discussion by dninomiya** [facebook/docusaurus#7133] [Q&A] [answered]
For example, I think it is possible if you have the following function.

```js
getDocTitle('docId(fileName)')
```

**Comment by slorber** [comment, marked as answer]
You might find these 2 hooks useful. 

```js
import {useDocsVersion,useDocById} from '@docusaurus/theme-common';

useDocById("xyz").title

useDocsVersion().docs.find(doc => doc.title = "XYZ title").id
```

But keep in mind they are for now internal APIs that we might do breaking changes on (altough it's unlikely the feature will totally disappear), so only use this wisely in your site code (ie avoid publishing an open-source plugin using those)


Note: these hooks will only work on components rendered on docs page (ie not on your homepage or blog posts). One id can be used in multiple docs versions so you have to be browsing a specific version for Docusaurus to know which to return.

**Reply by dninomiya** [reply]
Thanks for the answer!

**Reply by dninomiya** [reply]
@slorber 
Is there a way to get the document title from the page component...?

**Reply by slorber** [reply]
@dninomiya unfortunately not possible atm

for bundle size reason, all docs data are not available everywhere

Only a small subset is available globally through the global data API (id, path, sidebar): https://docusaurus.io/__docusaurus/debug/globalData

This upcoming feature might help you enrich this global data with additional fields: https://github.com/facebook/docusaurus/issues/6923

(not perfect as you'd need to duplicate the title frontmatter somehow, but better than nothing, and verbosity/duplication will be later fixed with a `createFrontMatter` api)

**Reply by mellson** [reply]
@slorber it seems like `useDocById`  has been removed in v3? Is there a new way to do something similar? In my use case I need to grab the description from another document.

update: found it, just needed to import it from here: `import { useDocById } from '@docusaurus/theme-common/internal';`

**Reply by Leonzon95** [reply]
Has this been removed on `3.5.2` ? 
Getting `Module '"@docusaurus/theme-common/internal"' has no exported member 'useDocById'.ts(2305)` 

The error I get when building 
```
        Error: Can't render static file for pathname "/category/my-doc/"
            at generateStaticFile (/Users/leo/Code/umbrella/node_modules/@docusaurus/core/lib/ssg.js:119:15)
            at runNextTicks (node:internal/process/task_queues:60:5)
            at process.processImmediate (node:internal/timers:454:9)
            at async /Users/leo/Code/umbrella/node_modules/p-map/index.js:57:22 {
          [cause]: TypeError: (0 , internal.useDocById) is not a function
              at CardLink (server.bundle.js:5057:34)
              at Uc (server.bundle.js:744610:44)
              at Xc (server.bundle.js:744612:253)
              at Z (server.bundle.js:744618:89)
              at Xc (server.bundle.js:744612:481)
              at Z (server.bundle.js:744618:89)
              at Yc (server.bundle.js:744621:98)
              at Xc (server.bundle.js:744613:145)
              at Z (server.bundle.js:744618:89)
              at Yc (server.bundle.js:744621:98)
        }
      ]
    }
    ```

**Reply by slorber** [reply]
Renamed, v3.5 release notes

https://docusaurus.io/blog/releases/3.5#other-changes

[#10313](https://github.com/facebook/docusaurus/pull/10313): Blog-related @docusaurus/theme-common/internal APIs have been moved to @docusaurus/plugin-content-blog/client.
[#10316](https://github.com/facebook/docusaurus/pull/10316): Docs-related @docusaurus/theme-common/internal APIs have been moved to @docusaurus/plugin-content-docs/client.
